src/kernel/checks.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_KERNEL_CHECKS_H
+#define BITCOIN_KERNEL_CHECKS_H
+
+#include <optional>
+
+namespace kernel {
+
+struct Context;
+
+enum class SanityCheckError {
+    ERROR_ECC,
```

Comment:
The word "error" is redundant in the enum name and in each enum value: 

```
SanityCheckError::ERROR_ECC
           ^^^^^  ^^^^^
```

Maybe strip the `ERROR_` prefix: `SanityCheckError::ECC`?